frontend/src/views/SecretMainPage/SecretMainPage.tsx

```diff
@@ -366,7 +394,7 @@ const SecretMainPageContent = () => {
       {!isRollbackMode ? (
         <>
           <ActionBar
-            environment={environment}
+            environment={currentWorkspace?.environments.find((env) => env.slug === environment)!}
```

Comment:
You could also just pass the slug and let the action bar do it by itself. Thus if the environment doesn't change this won't be triggered also.

Reply:
refactored as suggested